### ISSUE TITLE
Fix some coding standards issues

### ISSUE THREAD

**Issue by Zodiac1978** [roytanck/rt-plugin-report#6]
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/

For example:
- space before opening curly braces "{"
- add visibility for functions in the class (e.g. public)
- full stops at the end of inline comments
- use of escaped translation (esc_html__) functions instead of classic one "__()" if no HTML is used
- space after "If" or "Foreach" 
...

**Comment by roytanck** [comment]
Fixed some of these, but I'll leave the issue open for now.

While I agree that it helps when cooperating with other people from the community, WP's coding standards are not mandatory for 3rd party plugins, right?

**Comment by Zodiac1978** [comment]
> WP's coding standards are not mandatory for 3rd party plugins, right?

No, but the Tide project (https://make.wordpress.org/tide/ or https://github.com/wptide/wptide) will maybe add in the future some score which is showing if the WPCS are used or not. Meaning: the score is based on how much the WPCS are used or not.

**Comment by roytanck** [comment]
@Zodiac1978 Thank you for your pull request! Does it address enough of the issues to close this issue now?

**Comment by Zodiac1978** [comment]
> Does it address enough of the issues to close this issue now?

This decision is up to you :)

There some more issues ...

```
rt-plugin-report.php:
   1:1    error    phpcs:error    Class file names should be based on the class name with "class-" prepended. Expected class-rt-plugin-report.php, but found rt-plugin-report.php.
  13:2    error    phpcs:error    Missing @package tag in file comment
  23:5    error    phpcs:error    Missing doc comment for class RT_Plugin_Report
  25:16   error    phpcs:error    Missing member variable doc comment
  26:16   error    phpcs:error    Missing member variable doc comment
  27:16   error    phpcs:error    Missing member variable doc comment
  32:9    error    phpcs:error    Visibility must be declared on method "__construct"
  84:25   warning  phpcs:warning  Processing form data without nonce verification.
  85:43   warning  phpcs:warning  Processing form data without nonce verification.
  99:27   error    phpcs:error    All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'.
  99:94   error    phpcs:error    All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$version_temp'.
 102:108  error    phpcs:error    All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$wp_latest'.
 112:32   error    phpcs:error    All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'admin_url'.
 112:73   warning  phpcs:warning  Calling current_time() with a $type of "timestamp" or "U" is strongly discouraged as it will not return a Unix (UTC) timestamp. Please consider using a non-timestamp
                                  format or otherwise refactoring this code.
 136:26   error    phpcs:error    All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$this'.
 139:68   error    phpcs:error    All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$slug'.
 139:96   error    phpcs:error    All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$this'.
 153:6    error    phpcs:error    Doc comment for parameter "$hook" missing
 158:50   warning  phpcs:warning  Found: !=. Use strict comparisons (=== or !==).
 161:13   error    phpcs:error    Inline comments must end in full-stops, exclamation marks, or question marks
 162:13   warning  phpcs:warning  In footer ($in_footer) is not set explicitly wp_enqueue_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the
                                  footer, or explicitly `false` if it should be loaded in the header.
 162:13   error    phpcs:error    Resource version not set in call to wp_enqueue_script(). This means new versions of the script will not always be loaded due to browser caching.
 163:13   error    phpcs:error    Inline comments must end in full-stops, exclamation marks, or question marks
 172:13   error    phpcs:error    Resource version not set in call to wp_enqueue_style(). This means new versions of the style will not always be loaded due to browser caching.
 189:6    error    phpcs:error    Doc comment for parameter "$file" missing
 224:37   error    phpcs:error    Detected usage of a possibly undefined superglobal array index: $_POST['slug']. Use isset() or empty() to check the index exists before using it
 224:37   error    phpcs:error    $_POST data not unslashed before sanitization. Use wp_unslash() or similar
 240:18   warning  phpcs:warning  json_encode() is discouraged. Use wp_json_encode() instead.
 246:6    error    phpcs:error    Doc comment for parameter "$slug" missing
 264:55   warning  phpcs:warning  Found: ==. Use strict comparisons (=== or !==).
 308:6    error    phpcs:error    Doc comment for parameter "$report" missing
 315:26   warning  phpcs:warning  Found: ==. Use strict comparisons (=== or !==).
 315:26   error    phpcs:error    Use Yoda Condition checks, you must.
 329:55   warning  phpcs:warning  Found: !=. Use strict comparisons (=== or !==).
 335:79   warning  phpcs:warning  Calling current_time() with a $type of "timestamp" or "U" is strongly discouraged as it will not return a Unix (UTC) timestamp. Please consider using a non-timestamp
                                  format or otherwise refactoring this code.
 336:68   warning  phpcs:warning  Calling current_time() with a $type of "timestamp" or "U" is strongly discouraged as it will not return a Unix (UTC) timestamp. Please consider using a non-timestamp
                                  format or otherwise refactoring this code.
 351:6    error    phpcs:error    Doc comment for parameter "$message" missing
 359:6    error    phpcs:error    Doc comment for parameter "$available" missing
 359:6    error    phpcs:error    Doc comment for parameter "$optimal" missing
 364:98   warning  phpcs:warning  Found: ==. Use strict comparisons (=== or !==).
 368:127  warning  phpcs:warning  Found: ==. Use strict comparisons (=== or !==).
 376:6    error    phpcs:error    Doc comment for parameter "$perc" missing
 390:6    error    phpcs:error    Doc comment for parameter "$time_diff" missing
 413:39   warning  phpcs:warning  Found: ==. Use strict comparisons (=== or !==).
 413:39   error    phpcs:error    Use Yoda Condition checks, you must.
 417:36   warning  phpcs:warning  Found: ==. Use strict comparisons (=== or !==).
 417:36   error    phpcs:error    Use Yoda Condition checks, you must.
 425:6    error    phpcs:error    Doc comment for parameter "$compatibility" missing
 432:13   error    phpcs:error    Inline comments must end in full-stops, exclamation marks, or question marks
 447:6    error    phpcs:error    Doc comment for parameter "$version" missing
```

**Comment by roytanck** [comment]
Okay, let's leave it open for now and address all of them eventually.

**Comment by pedro-mendonca** [comment]
Hi @roytanck 
On #37 I've fixed some of the above list phpcs errors and warnings.

**Comment by roytanck** [comment]
Thank you, @pedro-mendonca . I've merged your PR and rolled out 1.8.2.

I think we can close this issue (for) now.
